appmap/pytest.py

```diff
@@ -1,20 +1,59 @@
 import pytest
 
-import appmap.unittest
 from appmap._implementation import testing_framework
+import appmap
+import appmap.wrapt as wrapt
+
+class recorded_testcase:
+    def __init__(self, item):
+        self.item = item
+
+    @wrapt.decorator
+    def __call__(self, wrapped, _, args, kwargs):
+        item = self.item
+        with item.session.appmap.record(
+                item.cls,
+                item.name,
+                method_id=item.originalname,
+                location=item.location):
+            wrapped(*args, **kwargs)
 
 
 if appmap.enabled():
     @pytest.hookimpl
     def pytest_sessionstart(session):
-        """Set up AppMap testing session and associate it with pytest and unittest sessions."""
-        # Use the same session for unittest in case there are any unittest.TestCases.
-        appmap.unittest.session = session.appmap = testing_framework.session(name='pytest', version=pytest.__version__)
+        session.appmap = testing_framework.session(name='pytest', version=pytest.__version__)
+
+    @pytest.hookimpl(hookwrapper=True)
```

Comment:
This doesn't need to be a hookwrapper; it doesn't do any cleanup. Just remove this decorator and the yield.